deprecation_notifier/DeprecationNotifier/DNAppDelegate.m

```diff
@@ -53,15 +64,38 @@ - (void)applicationDidFinishLaunching:(NSNotification *)aNotification {
       expectedVersionArray = [expectedVersionArray arrayByAddingObject:@"0"];
   }
   
-    
+  NSLog(@"Info: System version: %@ %@", systemVersion, systemBuild);
+  NSLog(@"Info: Expected version: %@ %@", expectedVersion, expectedBuildsStr);
   if (systemVersionArray.count < 3 || expectedVersionArray.count < 3) {
     NSLog(@"Exiting: Error, unable to properly determine system version or expected version");
     [NSApp terminate:nil];
+  } else if ([expectedVersionArray[0] intValue] < [systemVersionArray[0] intValue]) {
+    NSLog(@"Exiting: OS is already %@ or greater", expectedVersion);
+    [NSApp terminate:nil];
+  } else if (([expectedVersionArray[0] intValue] <= [systemVersionArray[0] intValue]) &&
+             ([expectedVersionArray[1] intValue] <  [systemVersionArray[1] intValue])) {
+    NSLog(@"Exiting: OS is already %@ or greater", expectedVersion);
+    [NSApp terminate:nil];
   } else if (([expectedVersionArray[0] intValue] <= [systemVersionArray[0] intValue]) &&
              ([expectedVersionArray[1] intValue] <= [systemVersionArray[1] intValue]) &&
-             ([expectedVersionArray[2] intValue] <= [systemVersionArray[2] intValue])) {
+             ([expectedVersionArray[2] intValue] <  [systemVersionArray[2] intValue])) {
     NSLog(@"Exiting: OS is already %@ or greater", expectedVersion);
     [NSApp terminate:nil];
+  } else if (([expectedVersionArray[0] intValue] == [systemVersionArray[0] intValue]) &&
+             ([expectedVersionArray[1] intValue] == [systemVersionArray[1] intValue]) &&
+             ([expectedVersionArray[2] intValue] == [systemVersionArray[2] intValue])) {
+    NSLog(@"Checking: OS is equal to %@, checking build version", expectedVersion);
+    if (expectedBuilds == nil) {
```

Comment:
Use `expectedBuilds.count == 0` or `!expectedBuilds.count` to catch both nil and an empty array